functions/__sf_section_vi_mode.fish

```diff
@@ -0,0 +1,47 @@
+#
+# Vi Mode
+#
+
+function __sf_section_vi_mode -d "Display vi mode status"
+	# ------------------------------------------------------------------------------
+	# Configuration
+	# ------------------------------------------------------------------------------
+
+	__sf_util_set_default SPACEFISH_VI_MODE_SHOW true
+	__sf_util_set_default SPACEFISH_VI_MODE_PREFIX " "
+	__sf_util_set_default SPACEFISH_VI_MODE_SUFFIX $SPACEFISH_PROMPT_DEFAULT_SUFFIX
+	__sf_util_set_default SPACEFISH_VI_MODE_INSERT "[I]"
```

Comment:
A small nitpick: When working with strings, we usually don't use quotes unless necessary; so that's either when we have whitespace within the string or when it can lead to buggy behavior without them (e.g. as arguments for `test`).

These default settings can likely work without the need for quotes. 🙂